File: Array/Intersection of Two Arrays II.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        
        dic = {}
        
        for n in nums1:
            if n in dic:
                dic[n] += 1
            else:
                dic[n] = 1
        
        intersection = list()
        
        for n in nums2:
            if n in dic and dic[n] > 0:
                intersection.append(n)
                dic[n] -= 1
        
        return intersection

File: Intersection of Two Arrays II.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> count(1001, 0);
        vector<int> ans;
        for (int n:nums1)
            count[n]++;
        for (int n:nums2)
            if (count[n])
            {
                ans.push_back(n);
                count[n]--;
            }
        return ans;
    }
};
